core/src/main/kotlin/event/automoderation/AutoModerationActionExecutionEvent.kt

```diff
@@ -0,0 +1,128 @@
+package dev.kord.core.event.automoderation
+
+import dev.kord.common.entity.Permission.ManageGuild
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.Kord
+import dev.kord.core.behavior.MemberBehavior
+import dev.kord.core.behavior.MessageBehavior
+import dev.kord.core.behavior.automoderation.TypedAutoModerationRuleBehavior
+import dev.kord.core.behavior.channel.GuildMessageChannelBehavior
+import dev.kord.core.entity.Guild
+import dev.kord.core.entity.Member
+import dev.kord.core.entity.Message
+import dev.kord.core.entity.automoderation.AutoModerationAction
+import dev.kord.core.entity.automoderation.AutoModerationRule
+import dev.kord.core.entity.automoderation.BlockMessageAutoModerationAction
+import dev.kord.core.entity.automoderation.SendAlertMessageAutoModerationAction
+import dev.kord.core.entity.channel.GuildMessageChannel
+import dev.kord.core.event.automoderation.data.AutoModerationActionExecutionEventData
+import dev.kord.core.supplier.EntitySupplier
+import dev.kord.core.supplier.EntitySupplyStrategy
+import dev.kord.gateway.Intent.MessageContent
+
+/**
+ * An [AutoModerationEvent] that is sent when an [AutoModerationRule] is triggered and an [AutoModerationAction] is
+ * executed (e.g. when a message is [blocked][BlockMessageAutoModerationAction]).
+ *
+ * This event is only sent to bot users which have the [ManageGuild] permission.
+ */
+public class AutoModerationActionExecutionEvent(
+    public val data: AutoModerationActionExecutionEventData,
+    override val kord: Kord,
+    override val shard: Int,
+    override val customContext: Any?,
+    override val supplier: EntitySupplier = kord.defaultSupplier,
+) : AutoModerationEvent {
+
+    /** The ID of the [Guild] in which the [action] was executed. */
+    override val guildId: Snowflake get() = data.guildId
+
+    /** The [AutoModerationAction] which was executed. */
+    public val action: AutoModerationAction get() = AutoModerationAction(data.action, kord)
+
+    /** The ID of the [AutoModerationRule] which the [action] belongs to. */
+    override val ruleId: Snowflake get() = data.ruleId
+
+    /** The behavior of the [AutoModerationRule] which the [action] belongs to. */
+    override val rule: TypedAutoModerationRuleBehavior
+        get() = TypedAutoModerationRuleBehavior(guildId, ruleId, data.ruleTriggerType, kord)
+
+    /** The ID of the [Member] which generated the content which triggered the rule. */
+    public val userId: Snowflake get() = data.userId
+
+    /** The behavior of the [Member] which generated the content which triggered the rule. */
+    public val user: MemberBehavior get() = MemberBehavior(guildId, id = userId, kord)
```

Comment:
Should these be named `memberId` and `member`?

Reply:
Yes, as rules make no sense outside guilds; and when you are dealing with guilds you are dealing with members

Reply:
Done